core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.util.List;
+import java.util.function.Function;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.logging.ContentPreviewerFactory;
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpServiceElement;
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpServiceFactory;
+import com.linecorp.armeria.server.annotation.ExceptionHandlerFunction;
+import com.linecorp.armeria.server.annotation.RequestConverterFunction;
+import com.linecorp.armeria.server.annotation.ResponseConverterFunction;
+import com.linecorp.armeria.server.logging.AccessLogWriter;
+
+/**
+ * A builder class for binding a {@link Service} fluently. This class can be instantiated through
+ * {@link ServerBuilder#annotatedService()}.
+ *
+ * <p>Call {@link #build(Object)} to build the {@link Service} and return to the {@link ServerBuilder}.
+ *
+ * <pre>{@code
+ * ServerBuilder sb = Server.builder();
+ * sb.annotatedService()                                    // Returns an instance of this class
+ *   .requestTimeoutMillis(5000)
+ *   .maxRequestLength(8192)
+ *   .exceptionHandler((ctx, request, cause) -> HttpResponse.of(400))
+ *   .pathPrefix("/foo")
+ *   .verboseResponses(true)
+ *   .contentPreview(500)
+ *   .build(new Service())                                   // Return to the ServerBuilder.
+ *   .build();
+ * }</pre>
+ *
+ * @see ServiceBindingBuilder
+ */
+public final class AnnotatedServiceBindingBuilder implements ServiceConfigSetters {
```

Comment:
Instead of extending `DefaultServiceConfigSetter` I thought of implementing `ServiceConfigSetters ` and delegating calls to `DefaultServiceConfigSetters `. Hope this is fine